From: Ubuntu Developers <ubuntu-devel-discuss@lists.ubuntu.com>
Date: Fri, 20 Dec 2019 16:45:51 +0100
Subject: _extrausers_deluser

===================================================================
---
 src/userdel.c | 23 +++++++++++++++++++++++
 1 file changed, 23 insertions(+)

Index: shadow-4.8.1/src/userdel.c
===================================================================
--- shadow-4.8.1.orig/src/userdel.c
+++ shadow-4.8.1/src/userdel.c
@@ -139,6 +139,12 @@ static int remove_mailbox (void);
 static int remove_tcbdir (const char *user_name, uid_t user_id);
 #endif				/* WITH_TCB */
 
+#ifndef EXTRAUSERS_OPT
+#define EXTRAUSERS_OPT 100000
+#endif
+
+static bool use_extrausers = false;
+
 /*
  * usage - display usage message and exit
  */
@@ -157,6 +163,7 @@ static void usage (int status)
 	(void) fputs (_("  -r, --remove                  remove home directory and mail spool\n"), usageout);
 	(void) fputs (_("  -R, --root CHROOT_DIR         directory to chroot into\n"), usageout);
 	(void) fputs (_("  -P, --prefix PREFIX_DIR       prefix directory where are located the /etc/* files\n"), usageout);
+	(void) fputs (_("      --extrausers              Use the extra users database\n"), usageout);
 #ifdef WITH_SELINUX
 	(void) fputs (_("  -Z, --selinux-user            remove any SELinux user mapping for the user\n"), usageout);
 #endif				/* WITH_SELINUX */
@@ -1035,6 +1042,7 @@ int main (int argc, char **argv)
 			{"remove",       no_argument,       NULL, 'r'},
 			{"root",         required_argument, NULL, 'R'},
 			{"prefix",       required_argument, NULL, 'P'},
+                        {"extrausers", no_argument,       NULL, EXTRAUSERS_OPT},
 #ifdef WITH_SELINUX
 			{"selinux-user", no_argument,       NULL, 'Z'},
 #endif				/* WITH_SELINUX */
@@ -1048,6 +1056,9 @@ int main (int argc, char **argv)
 #endif				/* !WITH_SELINUX */
 		                         long_options, NULL)) != -1) {
 			switch (c) {
+                        case EXTRAUSERS_OPT:
+                                use_extrausers = true;
+                                break;
 			case 'f':	/* force remove even if not owned by user */
 				fflg = true;
 				break;
@@ -1136,6 +1147,18 @@ int main (int argc, char **argv)
 	is_sub_gid = sub_gid_file_present ();
 #endif				/* ENABLE_SUBIDS */
 
+        if (use_extrausers) {
+               pw_setdbname (EXTRAUSERS_PASSWD_FILE);
+               spw_setdbname (EXTRAUSERS_SHADOW_FILE);
+               gr_setdbname (EXTRAUSERS_GROUP_FILE);
+               /* TODO expose this information in other tools */
+               sub_uid_setdbname(EXTRAUSERS_SUBUID_FILE);
+               sub_gid_setdbname(EXTRAUSERS_SUBGID_FILE);
+#ifdef SHADOWGRP
+               sgr_setdbname (EXTRAUSERS_SHADOWGROUP_FILE);
+#endif
+        }
+
 	/*
 	 * Start with a quick check to see if the user exists.
 	 */
